apps/costcontrol/js/views/telephony.js

```diff
@@ -112,8 +112,10 @@ var TelephonyTab = (function() {
       billingCycle.hidden = true;
     } else {
       billingCycle.hidden = false;
-      var content = Formatting.getFormattedDate(settings.nextReset,
-        _('short-date-format'));
+      var content = Formatting.getFormattedDate(settings.nextReset, {
```

Comment:
Ok, I cached the formatters and added code to reset them on language/timeformat changes.